scripts/logs-backup-from-all-pods/extract_logs.py

```diff
@@ -0,0 +1,187 @@
+        date = datetime.datetime.now() - datetime.datetime(1970, 1, 1)
+        date = date - datetime.timedelta(days=int(days))
+        seconds = (date.total_seconds())
+        epoch_start_date = round(seconds*1000)
+        # print("epoch_start_date: ", epoch_start_date)
+        for plKey, execution in pl_key_exec_dict.items():
+            exec_str = str(execution.decode("utf-8"))
+            field = "stage." + exec_str + ".endTime"
+            time = redis_conn.hget(plKey, field)
+            # if no time then consider start_time
+            if time is None:
+                field = "stage." + exec_str + ".startDate"
+                time = redis_conn.hget(plKey, field)
+                if time is None:
+                    # print("No start and end dates found for ", plKey)
+                    continue
+            time = str(time.decode("utf-8"))
+            time = int(time)
+            # print("The time is: ", time)
+            if epoch_start_date <= time:
+                # print("Found in time execution")
+                trigger_bytes = redis_conn.hget(plKey, "trigger")
+                trigger_str = str(trigger_bytes.decode("utf-8"))
+                trigger_json = json.loads(trigger_str)
+                name = trigger_json['user']
+                # print("and the user in time execution is: ", name)
+                names.add(name)
+    else:
+        for key in keys:
+            plKey = key[:35]
+            trigger_bytes = redis_conn.hget(plKey, "trigger")
+            trigger_str = str(trigger_bytes.decode("utf-8"))
+            trigger_json = json.loads(trigger_str)
+            name = trigger_json['user']
+            names.add(name)
+    # print("names: ", names)
+    return len(names)
+
+
+def get_no_of_active_users_count():
+    if days.strip() != 'None':
+        sql = "SELECT count(distinct  data->'auditData'-> 'source'->> 'name') FROM audit_events WHERE data->>'eventType' @@ ' AUTHENTICATION_SUCCESSFUL_AUDIT' AND created_at >= %s"
+        date = datetime.now() - timedelta(days=int(days))
+        cur_audit.execute(sql, [date])
+    else:
+        sql = "SELECT count(distinct  data->'auditData'-> 'source'->> 'name') FROM audit_events WHERE data->>'eventType' @@ ' AUTHENTICATION_SUCCESSFUL_AUDIT'"
+        cur_audit.execute(sql)
+    return cur_audit.fetchone()[0]
+
+
+def start_extraction():
+    try:
+        cookie = login_to_isd()
+        # get applications count
+        url = isd_gate_url + "/applications"
+        apps = get_response(url, cookie)
+        apps_count = len(apps)
+        # print("apps_count: ", apps_count)
+        pipelines_count = 0
+        for app in apps:
+            name = app['name']
+            url = isd_gate_url +  "/applications/" + name + "/pipelineConfigs"
+            pipelines = get_response(url, cookie)
+            pipelines_count = pipelines_count + len(pipelines)
+        # print("pipelines_count: ", pipelines_count)
+        cookie = login_to_isd()
```

Comment:
We have already logged in at line 105. then why do we need to do it again ?

Reply:
The above loop consumes more time when there are many pipelines so to avoid session timeout issues another login attempt is made here.